internal/checker/nodebuilderimpl.go

```diff
@@ -0,0 +1,2972 @@
+	isConstructorNode bool
+	symbolId          ast.SymbolId
+	nodeId            ast.NodeId
+}
+
+type TrackedSymbolArgs struct {
+	symbol               *ast.Symbol
+	enclosingDeclaration *ast.Node
+	meaning              ast.SymbolFlags
+}
+
+type SerializedTypeEntry struct {
+	node           *ast.Node
+	truncating     bool
+	addedLength    int
+	trackedSymbols []*TrackedSymbolArgs
+}
+
+type CompositeTypeCacheIdentity struct {
+	typeId        TypeId
+	flags         nodebuilder.Flags
+	internalFlags nodebuilder.InternalFlags
+}
+
+type NodeBuilderLinks struct {
+	serializedTypes                  map[CompositeTypeCacheIdentity]*SerializedTypeEntry // Collection of types serialized at this location
+	fakeScopeForSignatureDeclaration *string                                             // If present, this is a fake scope injected into an enclosing declaration chain.
+}
+
+type NodeBuilderSymbolLinks struct {
+	specifierCache module.ModeAwareCache[string]
+}
+type NodeBuilderContext struct {
+	tracker                         nodebuilder.SymbolTracker
+	approximateLength               int
+	encounteredError                bool
+	truncating                      bool
+	reportedDiagnostic              bool
+	flags                           nodebuilder.Flags
+	internalFlags                   nodebuilder.InternalFlags
+	depth                           int
+	enclosingDeclaration            *ast.Node
+	enclosingFile                   *ast.SourceFile
+	inferTypeParameters             []*Type
+	visitedTypes                    map[TypeId]bool
+	symbolDepth                     map[CompositeSymbolIdentity]int
+	trackedSymbols                  []*TrackedSymbolArgs
+	mapper                          *TypeMapper
+	reverseMappedStack              []*ast.Symbol
+	enclosingSymbolTypes            map[ast.SymbolId]*Type
+	suppressReportInferenceFallback bool
+	remappedSymbolReferences        map[ast.SymbolId]*ast.Symbol
+
+	// per signature scope state
+	hasCreatedTypeParameterSymbolList     bool
+	hasCreatedTypeParametersNamesLookups  bool
+	typeParameterNames                    map[TypeId]*ast.Identifier
+	typeParameterNamesByText              map[string]struct{}
+	typeParameterNamesByTextNextNameCount map[string]int
+	typeParameterSymbolList               map[ast.SymbolId]struct{}
+}
+
+type nodeBuilderImpl struct {
+	// host members
+	f  *ast.NodeFactory
+	ch *Checker
+	e  *printer.EmitContext
+
+	// cache
+	links       core.LinkStore[*ast.Node, NodeBuilderLinks]
+	symbolLinks core.LinkStore[*ast.Symbol, NodeBuilderSymbolLinks]
+
+	// closures
+	typeToTypeNodeClosure               func(t *Type) *ast.TypeNode
+	typeReferenceToTypeNodeClosure      func(t *Type) *ast.TypeNode
+	conditionalTypeToTypeNodeClosure    func(t *Type) *ast.TypeNode
+	createTypeNodeFromObjectTypeClosure func(t *Type) *ast.TypeNode
+	isStringNamedClosure                func(d *ast.Declaration) bool
+	isSingleQuotedStringNamedClosure    func(d *ast.Declaration) bool
```

Comment:
I would suggest inverting this and give these the "nice names", since they are what should be called by everyone.